kv/kv.go

```diff
@@ -13,6 +13,47 @@
 
 package kv
 
+import "bytes"
+
+// EncodedKey represents encoded key in low-level storage engine.
+type EncodedKey []byte
```

Comment:
We need to distinguish these two types. Key cannot be use as EncodedKey, although they are both []byte.